AEPMessaging/Sources/Messaging+EdgeEvents.swift

```diff
@@ -0,0 +1,220 @@
+/*
+ Copyright 2021 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPCore
+import AEPServices
+import Foundation
+
+extension Messaging {
+    // MARK: - internal methods
+
+    /// Sends an experience event to the platform SDK for tracking the notification click-throughs
+    ///
+    /// - Parameters:
+    ///   - event: The triggering event with the click through data
+    ///   - config: configuration data
+    /// - Returns: A boolean explaining whether the handling of tracking info was successful or not
+    func handleTrackingInfo(event: Event, _ config: [AnyHashable: Any]) {
+        guard let expEventDatasetId = config[MessagingConstants.SharedState.Configuration.EXPERIENCE_EVENT_DATASET] as? String, !expEventDatasetId.isEmpty else {
+            Log.warning(label: MessagingConstants.LOG_TAG,
+                        "Failed to handle tracking information for push notification: " +
+                            "Experience event dataset ID from the config is invalid or not available. '\(event.id.uuidString)'")
+            return
+        }
+
+        // Get the xdm data with push tracking details
+        guard var xdmMap = getXdmData(event: event, config: config) else {
+            Log.warning(label: MessagingConstants.LOG_TAG,
+                        "Failed to handle tracking information for push notification: " +
+                            "Error while creating xdmMap with the push tracking details from the event and config. '\(event.id.uuidString)'")
+            return
+        }
+
+        // Add application specific tracking data
+        let applicationOpened = event.applicationOpened
+        xdmMap = addApplicationData(applicationOpened: applicationOpened, xdmData: xdmMap)
```

Comment:
event.applicationOpened can be directly used..
no need of extra let 

```
 let applicationOpened = event.applicationOpened        xdmMap = addApplicationData(applicationOpened: applicationOpened, xdmData: xdmMap)
```

can be made

```
xdmMap = addApplicationData(applicationOpened: event.applicationOpened, xdmData: xdmMap)
```